File: Top 100 Codes/03-Codes for Recursion/04-Smallest element in an array/Smallestelementinanarray.py
def smallest_element(arr, n):
    """
    Recursive function to find the smallest element in an array.
    
    :param arr: List of integers
    :param n: Size of the list
    :return: Smallest element in the list
    """
    # Base case: If the list has only one element, return that element
    if n == 1:
        return arr[0]

    # Recursive case: Compare the last element with the smallest in the remaining list
    return min(arr[n - 1], smallest_element(arr, n - 1))


# Example usage
arr = [10, 45, 78, 34, 67]

# Find the smallest element
print("Smallest Element is", smallest_element(arr, len(arr)))

File: Top 100 Codes/03-Codes for Recursion/04-Smallest element in an array/Smallestelementinanarray.cpp
#include <bits/stdc++.h> // Standard header file for competitive programming
using namespace std;

int smallest_element(int n, int arr[]) {
    // Base case: If the array has only one element, return that element
    if (n == 1) 
        return arr[0];

    // Recursive case: Compare the last element with the smallest in the remaining array
    return min(arr[n - 1], smallest_element(n - 1, arr));
}

int main() {
    // Input array
    int arr[] = {10, 45, 78, 34, 67};
    
    // Calculate the size of the array
    int n = sizeof(arr) / sizeof(arr[0]);

    // Find and print the smallest element in the array
    cout << "Smallest Element is " << smallest_element(n, arr) << endl;

    return 0;
}
